auth/auth-core/src/main/java/com/bazaarvoice/emodb/auth/identity/AuthIdentity.java

```diff
@@ -26,16 +51,22 @@
     private Date _issued;
 
 
-    protected AuthIdentity(String id, Set<String> roles) {
+    protected AuthIdentity(String id, String internalId, Set<String> roles) {
         checkArgument(!Strings.isNullOrEmpty(id), "id");
+        checkArgument(!Strings.isNullOrEmpty(internalId), "internalId");
```

Comment:
I went with `!Strings.isNullOrEmpty(id)` because this is a stronger check then just not-null, and we shouldn't support either one being an empty string.